src/lcm/simulation/simulate.py

```diff
@@ -89,8 +85,29 @@ def simulate(
     # ----------------------------------------------------------------------------------
     simulation_results = {}
 
+    last_period = n_periods - 1
+
     for period in range(n_periods):
-        state_action_space = state_action_space.replace(states)
+        logger.info("Period: %s", period)
+
+        is_last_period = period == last_period
+
+        relevant_states = model.variable_info.query(
+            "is_state and enters_concurrent_valuation"
+        ).index.tolist()
+        if not is_last_period:
+            relevant_states += model.variable_info.query(
+                "is_state and enters_transition"
+            ).index.tolist()
+
+        states_for_state_action_space = {n: states[n] for n in set(relevant_states)}
+
+        state_action_space = create_state_action_space(
+            model=model,
+            initial_states=states_for_state_action_space,
```

Comment:
```suggestion
            current_states=states_for_state_action_space,
```
?

Looks a little like the naming might be a leftover of this being done once for all periods? More a question/reminder than request for implementation, though.